modules/images/webp-uploads/load.php

```diff
@@ -599,6 +599,23 @@ function webp_uploads_img_tag_update_mime_type( $image, $context, $attachment_id
 	 */
 	$target_mimes = apply_filters( 'webp_uploads_content_image_mimes', array( 'image/webp', 'image/jpeg' ), $attachment_id, $context );
 
+	/**
+	 * Filters whether the smaller image should be used regardless of which MIME type is preferred overall.
+	 *
+	 * This is disabled by default only because it is not part of the current WordPress core feature proposal, as it
+	 * requires storing the file sizes, which the plugin implementation does but the core implementation does not.
+	 *
+	 * By enabling this, the plugin will compare the image file sizes and prefer the smaller file regardless of MIME
+	 * type.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param bool $prefer_smaller_image_file Whether to prefer the smaller image file.
+	 */
+	if ( apply_filters( 'webp_uploads_prefer_smaller_image_file', false ) ) {
+		$target_mimes = webp_uploads_get_mime_types_by_filesize( $target_mimes, $attachment_id );
```

Comment:
probably should pass context here as well?

Reply:
@adamsilverstein That would certainly be possible, but I'm thinking that wouldn't make so much sense here since this filter here is to toggle that entire part of the behavior. I can't see why anyone would want to enable this in one place but not another. WDYT about that?

Reply:
yea, I'm not sure why you would want to do this in one context and not another, jut noticed it was missing. fine to leave off for now.